components/script/dom/blob.rs

```diff
@@ -70,23 +94,20 @@ impl Blob {
         Ok(Blob::new(global, bytes, &typeString.to_ascii_lowercase()))
     }
 
-    pub fn read_out_buffer(&self, send: Sender<Vec<u8>>) {
-        send.send(self.bytes.clone().unwrap_or(vec![])).unwrap();
-    }
+    pub fn get_bytes(&self) -> Vec<u8> {
+        let start = self.bytes_start.unwrap_or(0);
+        let end = self.bytes_end.unwrap_or(self.bytes.len());
 
-    // simpler to use version of read_out_buffer
-    pub fn clone_bytes(&self) -> Vec<u8> {
-        self.bytes.clone().unwrap_or(vec![])
+        self.bytes[start..end].to_vec()
```

Comment:
This (`.to_vec()`) still copies bytes. It looks like this method is used to send `Vec<u8>` messages, so it probably can’t simply return `&[u8]` instead.

To reduce copying, the `Vec<u8>` messages should probably be changed to some reference counted thing. Maybe tendrils? https://github.com/servo/tendril

CC @jdm